src/pages/patientView/mutationalSignatures/MutationalSignaturesContainer.tsx

```diff
@@ -1,34 +1,102 @@
-import { IMutationalSignature } from 'shared/model/MutationalSignature';
+import { Observer, observer } from 'mobx-react';
+import { computed, action, makeObservable, observable } from 'mobx';
+import {
+    IMutationalSignature,
+    IMutationalCounts,
+} from 'shared/model/MutationalSignature';
 import ClinicalInformationMutationalSignatureTable from '../clinicalInformation/ClinicalInformationMutationalSignatureTable';
 import Select from 'react-select';
-import { MolecularProfile } from 'cbioportal-ts-api-client';
+import autobind from 'autobind-decorator';
+
+import {
+    ClinicalDataBySampleId,
+    MolecularProfile,
+    Sample,
+} from 'cbioportal-ts-api-client';
 import {
     getVersionOption,
     getVersionOptions,
+    getSampleOptions,
+    getSampleOption,
+    MutationalSignaturesVersion,
 } from 'shared/lib/GenericAssayUtils/MutationalSignaturesUtils';
 import _ from 'lodash';
+import { ButtonGroup, Button } from 'react-bootstrap';
+
+import MutationalBarChart from 'pages/patientView/mutationalSignatures/MutationalSignatureBarChart';
+import SignatureTextBox from 'pages/patientView/mutationalSignatures/SignatureTextBox';
+import { getPercentageOfMutationalCount } from './MutationalSignatureBarChartUtils';
+import {
+    DefaultTooltip,
+    placeArrowBottomLeft,
+    DownloadControls,
+} from 'cbioportal-frontend-commons';
+import classNames from 'classnames';
 
 export interface IMutationalSignaturesContainerProps {
     data: { [version: string]: IMutationalSignature[] };
     profiles: MolecularProfile[];
-    onVersionChange: (version: string) => void;
     version: string;
+    sample: string;
+    samples: string[];
+    onVersionChange: (version: string) => void;
+    onSampleChange: (sample: string) => void;
+    dataCount: { [version: string]: IMutationalCounts[] };
+}
+
+interface IAxisScaleSwitchProps {
+    onChange: (selectedScale: AxisScale) => void;
+    selectedScale: AxisScale;
+}
+
+export enum AxisScale {
+    PERCENT = '%',
+    COUNT = '#',
 }
 
 @observer
 export default class MutationalSignaturesContainer extends React.Component<
     IMutationalSignaturesContainerProps,
     {}
 > {
+    @observable signatureProfile: string;
+    @observable.ref private plotSvg: SVGElement | null = null;
+    @observable signatureURL: string;
+    @observable signatureDescription: string;
+    @observable signatureInformationToolTipVisible: boolean = false;
+    public static defaultProps: Partial<IAxisScaleSwitchProps> = {
+        selectedScale: AxisScale.COUNT,
+    };
+
+    @observable
+    selectedScale: string = AxisScale.COUNT;
+
+    mutationalProfileSelection = (childData: string, visibility: boolean) => {
+        this.signatureProfile = childData;
+        this.signatureInformationToolTipVisible = visibility;
+        this.signatureURL = this.props.data[this.props.version].filter(obj => {
+            if (childData === obj.meta.name) {
+                return obj;
+            }
+        })[0].meta.url;
```

Comment:
Added the ternary statement that checks whether the output of the filter has a length >0, if yes it will return the `meta.description`, otherwise `No description available`. 

For the URL, it will return an empty string and in the tooltip we display `'No link to external website available'`.